### PR TITLE
Add error message when not using hipcc and  CMAKE_CXX_STANDARD is not set

### PR DESCRIPTION
**Pull request by Rombur**
When using `Cray clang` or `amdclang`, we need to set `CMAKE_CXX_STANDARD` if we want to use C++14. As reported in #5921, there is no problem when using C++17 and therefore no problem with Kokkos 4.0.

### CODE REVIEW

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -212,6 +212,9 @@ GLOBAL_SET(KOKKOS_AMDGPU_OPTIONS)
 IF(KOKKOS_ENABLE_HIP)
   SET(AMDGPU_ARCH_FLAG "--offload-arch")
   IF(NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
+    IF(KOKKOS_CXX_STANDARD STREQUAL 14 AND NOT CMAKE_CXX_STANDARD)
+      message(FATAL_ERROR "Set CMAKE_CXX_STANDARD to 14")
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:216]: The title line say "warning" but this is actually erroring out.  Did you mean to?

Comment (Rombur) [reply cmake/kokkos_arch.cmake:216]: I changed my mind after I created the PR and forgot to update the title.

**Review by dalg24 — APPROVED** [review]
Please update the tile line

**Review by nliber — APPROVED** [review]

**Comment by dalg24** [comment]
@Rombur please add to the changelog

**Pull request merged**